src/features/view-subdomains/SubdomainTable/useInfiniteSubdomains.ts

```diff
@@ -0,0 +1,109 @@
+import { useCallback, useEffect } from 'react';
+import { InfiniteData, useInfiniteQuery, useQueryClient } from 'react-query';
+
+import { useZnsSdk } from '../../../lib/hooks';
+import { getDomainId } from '../../../lib/util';
+import { DomainSortOptions } from '@zero-tech/zns-sdk/lib/api/dataStoreApi/types';
+
+/**
+ * Number of results to show per page.
+ * This is currently hardcoded, but could be made configurable in the future.
+ * Ideally different device types would have a different number of results
+ * per page for optimal performance.
+ */
+const DEFAULT_RESULTS_PER_PAGE = 24;
+
+const DEFAULT_SORT_OPTIONS: DomainSortOptions = {
+	name: 'asc',
+};
+
+export const getInfiniteSubdomainQueryKey = (parentDomainId: string) => {
+	return ['subdomains', { parentDomainId }];
+};
+
+/**
+ * Handles incrementally loading subdomains for a given domains.
+ * Leverages react-query's useInfiniteQuery hook to handle pagination.
+ * @param zna The domain to load subdomains for
+ */
+export const useInfiniteSubdomains = (zna: string) => {
+	const sdk = useZnsSdk();
+
+	const parentDomainId = getDomainId(zna);
+	const queryClient = useQueryClient();
+
+	// Assigning queryKey here as it's used in multiple places
+	const queryKey = getInfiniteSubdomainQueryKey(parentDomainId);
+
+	/**
+	 * Fetches the next page of subdomains from the data store
+	 */
+	const fetchSubdomains = useCallback(
+		async ({ pageParam = 0 }) => {
+			// Number of results already loaded, used to determine where to start the next query
+			const currentIndex = pageParam * DEFAULT_RESULTS_PER_PAGE;
+
+			console.log(parentDomainId);
+			console.log('currin', currentIndex, 'pageparam', pageParam);
+			sdk
+				.getSubdomainsById(parentDomainId, true, 24, 0, DEFAULT_SORT_OPTIONS)
```

Comment:
Would it make sense to name the hardcoded values e.g. `const resultsPerPage = 24` or something.

Oh not sure if this is to be removed as per comment below ...

Reply:
Yep removed this one!